src/storages/memory_storage/alter_table.rs

```diff
@@ -3,50 +3,169 @@
 use {
     super::MemoryStorage,
     crate::{
-        ast::ColumnDef,
-        result::{Error, MutResult},
-        store::AlterTable,
+        ast::ColumnDef, result::MutResult, schema::ColumnDefExt, store::AlterTable,
+        AlterTableError, Value,
     },
     async_trait::async_trait,
 };
 
+macro_rules! unwrap_or_return_err {
+    ($storage:expr, $expr:expr) => {
+        match $expr {
+            Ok(success) => success,
+            Err(err) => return Err(($storage, err.into())),
+        }
+    };
+}
+
 #[async_trait(?Send)]
 impl AlterTable for MemoryStorage {
-    async fn rename_schema(self, _table_name: &str, _new_table_name: &str) -> MutResult<Self, ()> {
-        Err((
-            self,
-            Error::StorageMsg("[MemoryStorage] alter-table is not supported".to_owned()),
-        ))
+    async fn rename_schema(self, table_name: &str, new_table_name: &str) -> MutResult<Self, ()> {
+        let mut storage = self;
+
+        match storage
+            .items
+            .remove(table_name)
+            .ok_or_else(|| AlterTableError::TableNotFound(table_name.to_owned()))
+        {
+            Ok(mut item) => {
+                item.schema.table_name = new_table_name.to_owned();
+                storage.items.insert(new_table_name.to_owned(), item);
+
+                Ok((storage, ()))
+            }
+            Err(err) => Err((storage, err.into())),
+        }
     }
 
     async fn rename_column(
         self,
-        _table_name: &str,
-        _old_column_name: &str,
-        _new_column_name: &str,
+        table_name: &str,
+        old_column_name: &str,
+        new_column_name: &str,
     ) -> MutResult<Self, ()> {
-        Err((
-            self,
-            Error::StorageMsg("[MemoryStorage] alter-table is not supported".to_owned()),
-        ))
+        let mut storage = self;
+
+        let item = unwrap_or_return_err!(
+            storage,
+            storage
+                .items
+                .get_mut(table_name)
+                .ok_or_else(|| AlterTableError::TableNotFound(table_name.to_owned()))
+        );
+
+        let column_def = unwrap_or_return_err!(
+            storage,
+            item.schema
+                .column_defs
+                .iter_mut()
+                .find(|column_def| column_def.name == old_column_name)
+                .ok_or(AlterTableError::RenamingColumnNotFound)
+        );
+        column_def.name = new_column_name.to_owned();
+
+        Ok((storage, ()))
```

Comment:
It looks that even though Try v2 has come into the stable Rust, we cannot replace `unwrap_or_return_err!` macro to the normal function which has exactly same functionality.
Then, it is not `try_..` which you mentioned is right.

Now, how about this?
We don't actually need `async` and `MutResult` for implementing internal logic.

If we add a new internal method for `AlterTable`, then we might not need this macro.

Let's suppose that we add a `rename_column_sync` function.
Then...
```rust
async fn rename_column(                                                             
    self,                                                                           
    table_name: &str,                                                               
    old_column_name: &str,                                                          
    new_column_name: &str,                                                          
) -> MutResult<Self, ()> {                                                          
    let mut storage = self;                                                         
                                                                                    
    match storage.rename_column_sync(table_name, old_column_name, new_column_name) {
        Ok(()) => Ok((storage, ())),                                                
        Err(e) => Err((storage, e)),                                                
    }                                                                               
}                                                                                   
```

```rust
impl MemoryStorage {                                                               
    fn rename_column_sync(                                                         
        &mut self,                                                                 
        table_name: &str,                                                          
        old_column_name: &str,                                                     
        new_column_name: &str,                                                     
    ) -> Result<()> {                                                              
        let item = self                                                            
            .items                                                                 
            .get_mut(table_name)                                                   
            .ok_or_else(|| AlterTableError::TableNotFound(table_name.to_owned()))?;
                                                                                   
        let mut column_def = item                                                  
            .schema                                                                
            .column_defs                                                           
            .iter_mut()                                                            
            .find(|column_def| column_def.name == old_column_name)                 
            .ok_or(AlterTableError::RenamingColumnNotFound)?;                      
                                                                                   
        column_def.name = new_column_name.to_owned();                              
                                                                                   
        Ok(())                                                                     
    }                                                                              
}                                                                                  
```


Like this above, actual implementation can even become more simple.

Reply:
There was '&mut self'.
I'm only thinking about avoiding mutable things, but I forgot that Storage allows it.
This is a good solution. I'll fix it right away

Reply:
How about making this a closure?

Reply:
MemoryStorage can be an example of a storage implementation.
So you have to write explicit code.
I will edit as per your feedback.

Reply:
Done.